src/logic/map_objects/immovable.cc

```diff
@@ -638,6 +639,10 @@ void Immovable::Loader::load(FileRead& fr, uint8_t const packet_version) {
 	} else {
 		imm.program_step_ = fr.signed_32();
 	}
+	imm.growth_delay_ =
+	   packet_version >= (packet_version > kCurrentPacketVersionImmovableNoFormerBuildings ? 9 : 8) ?
```

Comment:
Why do you have to manipulate the packet version itself?

Reply:
Am I overlooking some typo or did you overlook the logic magic ` >= ? > ? : :` ?

Reply:
No, I didn't. I was just wondering why it doesn't have its own packet version. I only looked at this diff though without loading the complete file to dig into it.

Reply:
For some reason Immovable has two packet versions (`kCurrentPacketVersionImmovable` and `kCurrentPacketVersionImmovableNoFormerBuildings`), and to find out whether we are loading a legacy savegame or not we need to figure out which of those versions `packet_version` refers to before we can decide which version constant to check for.